File: Python/Largest Perimeter Triangle.py
from typing import List


class Solution:
    """
    Time:   O(n*log(n))
    Memory: O(1)
    """

    def largestPerimeter(self, nums: List[int]) -> int:
        nums.sort(reverse=True)

        for i in range(len(nums) - 2):
            if nums[i + 2] + nums[i + 1] > nums[i]:
                return nums[i + 2] + nums[i + 1] + nums[i]

        return 0

File: Java/Largest Perimeter Triangle.java
import java.util.Arrays;

public class Solution {

    /**
     * Time:   O(n*log(n))
     * Memory: O(1)
     */
    public int largestPerimeter(int[] A) {
        Arrays.sort(A);

        for (int i = A.length - 3; i >= 0; --i)
            if (A[i] + A[i + 1] > A[i + 2])
                return A[i] + A[i + 1] + A[i + 2];

        return 0;
    }
}
